demos/multi_material/compositional_buoyancy/van_keken_1997.py

```diff
@@ -0,0 +1,281 @@
+# Rayleigh-Taylor instability
+# ===========================
+#
+# Rationale
+# ---------
+#
+# One may wish to simulate a geodynamical flow involving multiple physical phases. A
+# possible approach is to approximate phases as immiscible and forming a single fluid
+# whose dynamics can still be described as a single-phase Stokes flow. Under this
+# approximation, it is common to refer to immiscible phases as materials and to the
+# resulting simulations as multi-material. In such simulations, each material occupies
+# part of the numerical domain and is characterised by its own physical properties,
+# such as density and viscosity. Along material boundaries, physical properties are
+# averaged according to a chosen mathematical scheme.
+#
+# Numerical approach
+# ------------------
+#
+# To model the coexistence of multiple materials in the numerical domain, we employ an
+# interface-capturing approach called the conservative level-set method. Level-set
+# methods associate each material interface to a mathematical field representing a
+# measure of distance from that interface. In the conservative level-set approach, the
+# classic signed-distance function, $\phi$, employed in the level-set method is
+# transformed into a smooth step function, $\psi$, according to
+#
+# $$\psi(\mathbf{x}, t) = \frac{1}{2} \left[
+# \mathrm{tanh} \left( \frac{\phi(\mathbf{x}, t)}{2\epsilon} \right) + 1
+# \right]$$
+#
+# Throughout the simulation, the level-set field is advected with the flow:
+#
+# $$\frac{\partial \psi}{\partial t} + \nabla \cdot \left( \mathbf{u}\psi \right) = 0$$
+#
+# Advection of the level set deteriorates the shape of the initial profile. To maintain
```

Comment:
I find "deteriorates the shape of the initial profile" a little vague. Can you write explicitly what you mean here?